server/src/main/java/org/elasticsearch/transport/RemoteClusterService.java

```diff
@@ -248,15 +249,15 @@ private synchronized void updateSkipUnavailable(String clusterAlias, Boolean ski
     @Override
     protected void updateRemoteCluster(String clusterAlias, Settings settings) {
         CountDownLatch latch = new CountDownLatch(1);
-        updateRemoteCluster(clusterAlias, settings, ActionListener.runAfter(new ActionListener<Void>() {
+        updateRemoteCluster(clusterAlias, settings, ActionListener.runAfter(new ActionListener<>() {
             @Override
             public void onResponse(Void o) {
-                logger.debug("connected to new remote cluster {}", clusterAlias);
+                logger.debug("connected to new remote cluster [{}]", clusterAlias);
             }
 
             @Override
             public void onFailure(Exception e) {
-                logger.debug("connection to new remote cluster {} failed", clusterAlias);
+                logger.debug(new ParameterizedMessage("connection to new remote cluster [{}] failed: {}", clusterAlias, e.getMessage()), e);
```

Comment:
nit: you get the exception message with the stack trace, no need for a separate `e.getMessage()` here.